homeassistant/components/vodafone_station/button.py

```diff
@@ -0,0 +1,114 @@
+"""Vodafone Station buttons."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from typing import Any, Final
+
+from homeassistant.components.button import (
+    ButtonDeviceClass,
+    ButtonEntity,
+    ButtonEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import _LOGGER, DOMAIN
+from .coordinator import VodafoneStationRouter
+
+
+@dataclass
+class VodafoneStationBaseEntityDescriptionMixin:
+    """Mixin to describe a Button entity."""
+
+    press_action: Callable[[VodafoneStationRouter], Any]
+    is_suitable: Callable[[dict], bool]
+
+
+@dataclass
+class VodafoneStationEntityDescription(
+    ButtonEntityDescription, VodafoneStationBaseEntityDescriptionMixin
+):
+    """Vodafone Station entity description."""
+
+
+BUTTON_TYPES: Final = (
+    VodafoneStationEntityDescription(
+        key="reboot",
+        translation_key="reboot",
```

Comment:
This one can be removed as the device class gives the translation :)

Reply:
Yeah, noticed right after the commit ;-)